routers/repo/pull.go

```diff
@@ -568,7 +569,8 @@ func ParseCompareInfo(ctx *context.Context) (*models.User, *models.Repository, *
 	// format: <base branch>...[<head repo>:]<head branch>
 	// base<-head: master...head:feature
 	// same repo: master...feature
-	infos := strings.Split(ctx.Params("*"), "...")
+	infoPath, _ := url.QueryUnescape(ctx.Params("*"))
```

Comment:
I'd prefer the error in this case was handled - we can't know for sure that the parameter will be valid. So

```go
infoPath, err := url.QueryUnescape(ctx.Params("*"))
if err != nil {
  ctx.Handle(404, "QueryUnescape", err)
}
```

Reply:
ok